server/src/main/java/org/apache/iotdb/db/mpp/execution/memory/MemoryPool.java

```diff
@@ -169,32 +178,29 @@ public Pair<ListenableFuture<Void>, Boolean> reserve(
     }
 
     ListenableFuture<Void> result;
-    synchronized (this) {
-      if (maxBytes - reservedBytes < bytesToReserve
-          || maxBytesCanReserve
-                  - queryMemoryReservations
-                      .getOrDefault(queryId, Collections.emptyMap())
-                      .getOrDefault(fragmentInstanceId, Collections.emptyMap())
-                      .getOrDefault(planNodeId, 0L)
-              < bytesToReserve) {
-        LOGGER.debug(
-            "Blocked reserve request: {} bytes memory for planNodeId{}",
-            bytesToReserve,
-            planNodeId);
-        result =
-            MemoryReservationFuture.create(
-                queryId, fragmentInstanceId, planNodeId, bytesToReserve, maxBytesCanReserve);
-        memoryReservationFutures.add((MemoryReservationFuture<Void>) result);
-        return new Pair<>(result, Boolean.FALSE);
-      } else {
-        reservedBytes += bytesToReserve;
-        queryMemoryReservations
-            .computeIfAbsent(queryId, x -> new HashMap<>())
-            .computeIfAbsent(fragmentInstanceId, x -> new HashMap<>())
-            .merge(planNodeId, bytesToReserve, Long::sum);
-        result = Futures.immediateFuture(null);
-        return new Pair<>(result, Boolean.TRUE);
-      }
+    long tryRemainingBytes = remainingBytes.addAndGet(-bytesToReserve);
```

Comment:
It seems that the codes of logic of try-reserving memory and releasing memory have been repeated many times, it should be refined to two private methods, e.g.

```java
private boolean tryReserve() {
  long tryRemainingBytes = remainingBytes.addAndGet(-bytesToReserve);
  long queryRemainingBytes =
        maxBytesCanReserve
            - queryMemoryReservations
                .computeIfAbsent(queryId, x -> new ConcurrentHashMap<>())
                .computeIfAbsent(fragmentInstanceId, x -> new ConcurrentHashMap<>())
                .merge(planNodeId, bytesToReserve, Long::sum);
    return tryRemainingBytes >= 0 && queryRemainingBytes >= 0;
}

private void rollbackReserve() {
      queryMemoryReservations
          .computeIfAbsent(queryId, x -> new ConcurrentHashMap<>())
          .computeIfAbsent(fragmentInstanceId, x -> new ConcurrentHashMap<>())
          .merge(planNodeId, -bytesToReserve, Long::sum);   
   remainingBytes.addAndGet(bytesToReserve);
}
```